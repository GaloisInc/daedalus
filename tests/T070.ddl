def Main = 1 + 2

def f x = x.l


def P x = block
  x as int
  x as? int
  x as! int

def J x y = x # y

def L x =
  block
    for (s = 0; i in x) s + i
    map (i in x) true
