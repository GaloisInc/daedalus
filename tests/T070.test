show-types
T070.ddl
