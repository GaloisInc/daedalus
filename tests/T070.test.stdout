[WARNING] ./T070.ddl:7:3--7:10: Statement has no effect.
def Main:
  for any type a:
    `a` contains the number 1
    `a` contains the number 2
    `a` supports arithmetic
  defines: parser of a
   
def f:
  for any type a b:
    `a` has a field l : b
  parameter: a
  defines: b
   
def P:
  for any type a:
    `a` is coercible to `int`
    `a` is coercible to `int` (dynamic check)
    `a` is coercible to `int` (lossy)
  parameter: a
  defines: parser of int
   
def J:
  for any type a b c:
    a + b = c
  parameter: uint a
  parameter: uint b
  defines: parser of uint c
   
def L:
  for any type a b c d:
    `c` contains the number 0
    `a` is a collection with index `b`
    `a` is a collection of `c`
    `a` supports iteration
    `c` supports arithmetic
    `d` is a collection with index `b`
    `d` is a collection of `bool`
    mapping over `a` results in `d`
  parameter: a
  defines: parser of d
   
